### ISSUE TITLE
Make a quickstart maven goal

### ISSUE THREAD

**Issue by janbartel** [eclipse/jetty.project#1542]
It would be nice to be able to generate the quickstart-web.xml as part of a webapp module build, so that it is included in the packed war, but will be regenerated automatically every time the war is rebuilt.

**Comment by janbartel** [comment]
@ffissore  would be great if you have time to contribute to this.  You should be able to add another goal to the existing maven-jetty-plugin:  https://github.com/eclipse/jetty.project/tree/jetty-9.4.x/jetty-maven-plugin

**Comment by ffissore** [comment]
I'm sorry to say that I'm struggling a lot with this plugin: I cannot find a way to figure out or generate the correct classpath to run PreconfigureQuickStartWar.

My last attempt was using **part** of the command line printed by `java -jar start.jar --dry-run`, until `org.eclipse.jetty.xml.XmlConfiguration`, adding `jetty-quickstart` jar to the list, and finally PreconfigureQuickStartWar: `quickstart-web.xml` was generated **but** param `org.eclipse.jetty.tlds` was empty ( = `<![CDATA[]]>`).
However, using the same jetty distribution and generating quickstart-web.xml at webapp startup, `org.eclipse.jetty.tlds` listed the TLDs found in `${jetty.home.uri}/lib/apache-jstl/org.apache.taglibs.taglibs-standard-impl-1.2.5.jar`

On second thought, I think pre-generating the xml at `mvn package` time may not be a good idea. Output depends on the actual jetty configuration: making sense of such a maven plugin would require devs to have a local jetty distribution configured the same way of the production one.
Better would be to make quickstart an option of start.jar (or something like that), in order to generate quickstart-web.xml when the docker custom image is built.

What do you think?

**Comment by janbartel** [comment]
@ffissore I was thinking that we'd use one of the existing maven goals as the basis of the new plugin, ie AbstractJettyMojo.We would need to get the quickstart-web.xml parameterised with ${jetty.home} and ${jetty.base} when referring to jars, just like when it is generated by the distro. We could probably set a value for jetty.base to be the webapp that is being assembled in target, but jetty.home is a bit difficult. I guess we could post-process the quickstart-web.xml and if the jar isn't in the webapp or on some extraclasspath, then it must come from the container and thus we substitute ${jetty.home} for its url.

But your point is a good one that it all might depend on having a jetty distro locally that is configured the way the webapp expects.  We also have another issue open for making a plugin that can obtain all the jetty jars from a local distro rather than maven - see issue #741 - maybe we need to make that plugin before we can make a quickstart plugin?
